packages/server-wallet/src/wallet/index.ts

```diff
@@ -241,3 +257,11 @@ const takeActions = async (channels: Bytes32[]): Promise<ExecutionResult> => {
 
   return {outbox, error, channelResults};
 };
+
+function getOrThrow<E, T>(result: Either.Either<E, T>): T {
+  return Either.getOrElseW<E, T>(
+    (err: E): T => {
+      throw err;
+    }
+  )(result);
+}
```

Comment:
My only hesitation with the existence of this helper is that we don't really want to encourage using it. It might be simpler to implement some kind of catch all error response type out of the wallet, like `WalletError.unspecifiedError({ code: 500, msg: ... })` and log the exception with stack trace using `logger.error`?

I guess I'm just saying it's more valuable to return an unspecified left value to the wallet API caller than it is to throw it, since it's not like this is an _unexpected error_, those of which we do want to see things get thrown, this is expected but not handled. Dunno

Reply:
So, we probably _don't_ want to throw an error. Instead, we _want_ to return a strictly typed error in an error case.

This helper method is simply a fill-in until we decide what the wallet API should be, considering the changes that the run-loop architecture and the HTTP-friendly protocols force on us.

Reply:
Yeah my point is just that a slightly better fill-in could be to return some super generic typed error, as a value, to the API caller. As opposed to throwing the error. So, by throwing (using the current code), we would have:

```ts
try {
 {outgoing, channelResults} = wallet.doSomething()
} catch (e) {
  // e is either a known error, like UpdateChannelError or some unknown unknown bug
}
```

As opposed to with a generic error:

```ts
try {
 {errors, outgoing, channelResults} = wallet.doSomething()
} catch (e) {
  // e is some unknown unknown bug
}
```

Reply:
> Maybe out of scope for the PR, but I saw `getOrThrow` and was worried if we add this then within the next 7 days we might see it littered in 5 other places out of convenience.

I think that's a legitimate worry, but I believe we are definitely committed to having API methods return typed errors, and never throw. So it's fair to say we will remove this snippet within 7 days (ideally, by tomorrow!).

I'll add a note that it should not be used elsewhere.